### PR TITLE
feat: override content-type

### LINKED ISSUE #2121: [bug]: cannot override Content-Type
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

I need to send a WSDL request as `text/xml`, but that's not available from the drop-down types, so I chose `application/xml` and tried overriding it in the Headers section. The request is still sent as `application/xml` and not with the header I requested.

### Steps to reproduce

1. create a POST request
2. under "body" select `application/xml` and paste an XML content
3. under "headers" add `Content-Type: text/xml`
4. send request
5. check Network to see that `application/xml` was still used as a type

### Environment

Production

### Version

Cloud

### PR DESCRIPTION
**Pull request by anwarulislam**
Closes #2121 

### Description

Overriding `Content-Type` in Headers.

### Checks

- [x] My pull request adheres to the code style of this project
- [ ] My code requires changes to the documentation
- [ ] I have updated the documentation as required
- [x] All the tests have passed

### Additional Information

Previously, we were automatically assigning `Content-Type` from the body. No one could able to override the `Content-Type` from the header if they needed. They could be able to write the `Content-Type` but that wouldn't be prioritized. In this PR we basically fixed the issue. Checking if there's any `Content-Type` in headers then we're not adding Request Body's content-type with the request. Rather, the `Content-Type` from the header is being added in the actual request.

### CODE REVIEW

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
Please look into the comments for some fixes.

Also, I am not really satisfied with the behavior right now for multiple reasons.

1. If I press override and a valid override is applied, it would be cool if the button indicated that, the content-type is being overriden to this value, instead of just being an "Override" button.
2. If I create an override (which adds the header), then I remove the "Content-Type" header, clicking Override again doesn't do anything. Something would be nice there.

@liyasthomas can you chip in on the expected Override button behaviour as well ?

File: packages/hoppscotch-app/components/http/Body.vue [hunk 1]
```diff
@@ -80,19 +91,45 @@
...
+  restHeaders$,
+  setRESTContentType,
+  setRESTHeaders,
+} from "~/newstore/RESTSession"
 
-export default defineComponent({
-  setup() {
-    return {
-      validContentTypes: Object.keys(knownContentTypes),
+const emit = defineEmits(["changeTab"])
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/http/Body.vue:107]: Vue for TypeScript has an alternative way to define emits and props which tell the proper type signature of the emit functions.
Please check out: https://vuejs.org/api/sfc-script-setup.html#typescript-only-features

File: packages/hoppscotch-app/components/http/Body.vue [hunk 2]
```diff
@@ -80,19 +91,45 @@
...
 
-      contentType: useStream(restContentType$, null, setRESTContentType),
-    }
+const validContentTypes = Object.keys(knownContentTypes)
+const contentType = useStream(restContentType$, null, setRESTContentType)
+
+// The functional headers list (the headers actually in the system)
+const headers = useStream(restHeaders$, [], setRESTHeaders) as Ref<
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/http/Body.vue:113]: The type casting here is redundant.

File: packages/hoppscotch-app/components/http/Body.vue [hunk 3]
```diff
@@ -80,19 +91,45 @@
...
+  HoppRESTHeader[]
+>
+
+const overridenContentType = ref<string | null>(null)
+
+watch(
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/http/Body.vue:119]: Can't `overridenContentType` just be a computed property ?
Also, during the rewrite, try using `fp-ts` to build the result.

File: packages/hoppscotch-app/components/http/Body.vue [hunk 4]
```diff
@@ -49,6 +49,17 @@
             />
           </div>
         </tippy>
+        <ButtonSecondary
+          v-tippy="{ theme: 'tooltip' }"
+          :title="
+            overridenContentType
+              ? 'The Content-Type is being overriden to ' + overridenContentType
```
Comment (AndrewBastin) [inline packages/hoppscotch-app/components/http/Body.vue:56]: Is this i18n friendly ?

cc: @liyasthomas :eyes:

**Review by AndrewBastin — CHANGES_REQUESTED** [review]
Please look into the given comments.

File: packages/hoppscotch-app/components/http/Body.vue [hunk 5]
```diff
@@ -80,19 +99,63 @@
...
+const headers = useStream(restHeaders$, [], setRESTHeaders)
+
+const overridenContentType = ref<string>("")
+
+watch(
+  headers,
+  (allHeaders: HoppRESTHeader[]) => {
+    overridenContentType.value = pipe(
+      allHeaders,
+      A.findFirst((h) => h.key.toLowerCase() === "content-type"),
```
Comment (liyasthomas) [inline packages/hoppscotch-app/components/http/Body.vue:136]: `overridenContentType` should check the `active` boolean value in headers entry.

Comment (liyasthomas) [reply packages/hoppscotch-app/components/http/Body.vue:136]: Please use `findLast()`
```suggestion
      A.findLast((h) => h.key.toLowerCase() === "content-type"),
```
Last occurance has highest preference if there're multiple entries with same key.

File: packages/hoppscotch-app/components/http/Body.vue [hunk 6]
```diff
@@ -80,19 +99,56 @@
...
+const validContentTypes = Object.keys(knownContentTypes)
+const contentType = useStream(restContentType$, null, setRESTContentType)
+
+// The functional headers list (the headers actually in the system)
+const headers = useStream(restHeaders$, [], setRESTHeaders)
+
+const overridenContentType = computed(() =>
+  pipe(
+    headers.value,
+    A.findLast((h) => h.key.toLowerCase() === "content-type"),
```
Comment (liyasthomas) [inline packages/hoppscotch-app/components/http/Body.vue:131]:
```suggestion
    A.findLast((h) => h.key.toLowerCase() === "content-type" && h.active),
```
`overridenContentType` should check the active boolean value in the headers entry. cc: @anwarulislam

**Review by liyasthomas — CHANGES_REQUESTED** [review]

**Review by liyasthomas — APPROVED** [review]
LGTM.

**Pull request merged**